rewriter/ChainedSig.cc

```diff
@@ -0,0 +1,184 @@
+#include "rewriter/ChainedSig.h"
+#include "ast/Helpers.h"
+#include "ast/treemap/treemap.h"
+#include "core/errors/rewriter.h"
+
+using namespace std;
+namespace sorbet::rewriter {
+
+struct ChainedSigWalk {
+    // The previousSigSend is used to allow us to add errors to `sig` with missing block wihtout accidentally adding it
+    // to a chained sig. E.g.:
+    // `sig` -> error no block
+    // `sig.final {}` -> no error. Need to prevent adding the error when we receive `sig` by itself
+    ast::Send *previousSigSend;
```

Comment:
Do you mind resetting this to `nullptr` in `postTransformSend`?

That way, an invariant is that if we're currently transforming a sig send chain, it's set, but if we're transforming a completely unrelated (chain) of sigs, it's `nullptr`.

Reply:
hm, actually on second thought, it's not quite clear to me that this is straightforward to do, so maybe it's not worth doing. i'll think about it some more.